Generated client not compiled when x-go-name is specified

## Problem statement

When using abbreviations (in my case PDF) and x-go-name, non-compiled code is generated.

Example: a parameter named **pdf_extraction_mode** (translated to a literal _PdfExtractionMode_) and its x-go-name **PDFExtractionMode**.

The code in the *Params structure is generated correctly (the name is the same as in x-go-name), however, in the generated SetDefaults code, the _PdfExtractionMode_ literal is substituted, which breaks the compilation.

```
$ go build ./client/                                                                    
client/operations/extract_pdf_parameters.go:94:3: unknown field 'PdfExtractionMode' in struct literal of type ExtractPdfParams
```

## Swagger specification

```yaml
swagger: "2.0"
info:
  title: Sample API
  description: API description in Markdown.
  version: 1.0.0
host: api.example.com
basePath: /v1
schemes:
  - https
paths:
  /pdf:
    get:
      operationId: ExtractPdf
      parameters:
        - name: pdf_extraction_mode
          type: string
          enum:
            - fast
            - full
          x-go-name: PDFExtractionMode
          default: fast
          description: Lorem ipsum
          in: formData
      responses:
        200:
          description: OK
```

## Steps to reproduce

1. swagger generate client -q -A buggygen --spec example.yaml
2. trying to build this code (go build ./client)
 
## Environment
swagger version: 0.30.3  
go version: 1.19  
OS: MacOS Catalina

--- a/generator/templates/client/parameter.gotmpl
+++ b/generator/templates/client/parameter.gotmpl
@@ -140,7 +140,7 @@ func ({{ .ReceiverName }} *{{ pascalize .Name }}Params) SetDefaults() {
   val := {{ pascalize .Name }}Params{
   {{- range .Params }}
     {{- if and .HasDefault (not .IsFileParam) (not .IsStream) (not .IsBodyParam) }}
-    {{ pascalize .Name}}: {{ if and (not .IsArray) (not .IsMap) (not .HasDiscriminator) (or .IsNullable  ) }}&{{ end }}{{ varname .ID }}Default,
+    {{ pascalize .ID }}: {{ if and (not .IsArray) (not .IsMap) (not .HasDiscriminator) (or .IsNullable  ) }}&{{ end }}{{ varname .ID }}Default,
     {{- end }}
   {{- end }}
   }

